build_tools/benchmarks/common/benchmark_driver.py

```diff
@@ -27,19 +29,21 @@ def __init__(self,
     self.benchmark_suite = benchmark_suite
     self.benchmark_grace_time = benchmark_grace_time
     self.verbose = verbose
-    self.finished_benchmarks: List[Tuple[BenchmarkInfo, pathlib.Path]] = []
+    self.finished_benchmarks: List[pathlib.Path] = []
     self.finished_captures: List[pathlib.Path] = []
     self.benchmark_errors = []
     self._seen_benchmark_names: Set[str] = set()
 
-  def run_benchmark_case(self, benchmark_case: BenchmarkCase,
+  def run_benchmark_case(self, benchmark_info: BenchmarkInfo,
+                         benchmark_case: BenchmarkCase,
                          benchmark_results_filename: Optional[pathlib.Path],
                          capture_filename: Optional[pathlib.Path]) -> None:
-    """Runs the benchmark case and returns the results.
+    """Runs the benchmark case and serializes the results.
 
     Args:
+      benchmark_info: the benchmark info.
       benchmark_case: the benchmark_case.
```

Comment:
Neither of these have all of the fields of the other, but if it's reasonable to rewrite `BenchmarkInfo` to include the `BenchmarkCase` as a field then that could be less cumbersome.

Reply:
`BenchmarkInfo` is the common information of benchmark run to describe the devices.

Instead of passing `BenchmarkInfo` into `self.run_benchmark_case`, I'm thinking if we should only return `context`, `cpu_time`, `real_time` from `self.run_benchmark_case` (maybe pack them into a single result object). And still build the `BenchmarkRun` here with `BenchmarkInfo` and the result object.

Reply:
Since `run_benchmark_case` subsequently executes `__run_capture`, we'll want to construct something serializable in `__run_benchmark` in case `__run_capture` fails. Returning the results will also get significantly messier once we start adding other metrics (like host and device memory usage). I decided to name the result object `BenchmarkMetrics` to avoid confusion with `BenchmarkResults`.